Issue 5351 - author DC improvements

myroslavve opened the pull request:
# Description
Added missing fields to "author" in DC, and created "customer" type for fields from issue that are added by WooCommerce.
Added "current" option both for authors and customers:
For "author" it shows the author of the current post, and for "customer" it shows the user that is viewing the page.
<!---
Please include a summary of the changes and the related issue.
Please also include relevant motivation and context.
--->

Fixes #5351 

# How Has This Been Tested?
Checked that all new fields work both on frontend and editor. Checked that current option works, and also checked it on FSE.
<!---
Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration.
--->

# Test checklist

<!--- Please remove the unnecessary checkbox --->

**_ Front/Back Testing _**

-   [ ] Check that all new fields work both on frontend and editor.
-   [ ] Check that current option works, also check it on FSE.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced dynamic content rendering with new link targets for authors and customers.
	- Introduced a new property in the `LinkControl` component to disable the "Open in new tab" toggle.
	- Added a new method for responsive class updates in the `MaxiBlockComponent`.

- **Bug Fixes**
	- Improved handling of user data retrieval and link generation logic.

- **Style**
	- Added a new modifier class for disabled toggle switches to improve accessibility.

- **Documentation**
	- Updated constants for user-related links and attributes.

- **Chores**
	- General code optimizations and refactoring for improved readability and maintainability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

myroslavve committed:
Add missing fields to `user` type and add `customer` type for WC fields

myroslavve committed:
Add 'current' option to author and customer types

myroslavve committed:
Make 'current' option work for users on frontend

elzadj commented:
@myroslavve also pls check this one https://github.com/maxi-blocks/maxi-blocks/issues/5691#issuecomment-2304908093

myroslavve commented:
> @myroslavve also pls check this one [#5691 (comment)](https://github.com/maxi-blocks/maxi-blocks/issues/5691#issuecomment-2304908093)

Should be fixed here :+1:

elzadj commented on src/extensions/DC/getDCContent.js:
```diff
@@ -142,6 +142,20 @@ const getDCContent = async (dataRequest, clientId) => {
 		return getCartContent(dataRequest, data);
 	}
 
+	if (type === 'customers') {
+		const getCustomerDataField = field => data?.customerData?.[field]?.[0];
+
+		if (field === 'billing_name' || field === 'shipping_name') {
+			const firstNameField = field.replace('_name', '_first_name');
+			const firstName = getCustomerDataField(firstNameField);
+			const lastNameField = field.replace('_name', '_last_name');
+			const lastName = getCustomerDataField(lastNameField);
+			return `${firstName} ${lastName}`;
+		}
+
+		return getCustomerDataField(field) || null;
+	}
```
That makes sense to do, yep.

myroslavve committed:
Handle empty first or last customer
name correctly

Olekrut reviewed (changes requested):
Simple editor page and FSE were created on master, 
[simple editor.txt](https://github.com/user-attachments/files/16837995/simple.editor.txt)
[FSE code.txt](https://github.com/user-attachments/files/16837998/FSE.code.txt)


1) email - guess DC link should be mailto: current email, not author link
    website - guess, DC link should be to current website, not author link
    ![image](https://github.com/user-attachments/assets/b371237a-a24f-460b-a07e-ebaefe4c566e)
2) Message on frontend ( FSE )
![image](https://github.com/user-attachments/assets/ad4ba8df-20bd-4af3-b9ed-7578c12a3aa3)
3) Is should be current in simple editor?
![image](https://github.com/user-attachments/assets/f5ff2921-3065-4e75-9b3b-220f82fe24bc)
4) Maybe better to change label to Biographical info or Bio? ( like in user )
![image](https://github.com/user-attachments/assets/0a87c1b4-18a6-4941-bc2b-354c96687b50)

elzadj commented:
> website - guess, DC link should be to current website, not author link
Should be to the website from the field, but maybe it's better to add it as an option. 
Same for the email.

Also about the email - we need to at least  add the antispambot function for frontend (it's not a great one, but at least something) https://developer.wordpress.org/reference/functions/antispambot/

myroslavve committed:
Small fixes

Changed label and fixed php warning

myroslavve committed:
Update eslint no plusplus rule

myroslavve committed:
Add email and site link options to DC users

myroslavve committed:
Add warning about email scraping bots

myroslavve committed:
Merge branch 'master' into issue-5351

myroslavve committed:
Refactor: do async actions in parallel

myroslavve committed:
Refactor: fix last implementation

myroslavve committed:
Merge branch 'issue-5351' of https://github.com/yeahcan/maxi-blocks into issue-5351

myroslavve committed:
Fix FSE error when opening DC settings

myroslavve commented:
> 1. email - guess DC link should be mailto: current email, not author link
>    website - guess, DC link should be to current website, not author link

Added email and site options to link settings.

> 2. Message on frontend ( FSE )

Fixed.

> 3. Is should be current in simple editor?
>    ![image](https://private-user-images.githubusercontent.com/88893350/363701800-f5ff2921-3065-4e75-9b3b-220f82fe24bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3NDE0NzksIm5iZiI6MTcyNTc0MTE3OSwicGF0aCI6Ii84ODg5MzM1MC8zNjM3MDE4MDAtZjVmZjI5MjEtMzA2NS00ZTc1LTliM2ItMjIwZjgyZmUyNGJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA3VDIwMzI1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwOGNhNTM0M2I5ZTZiYzlhYjJhZTc2ZTVhYTAyMTYzOGU1NTUzOTZlMGJkYTYzODZjYTNiYWE0ZWQyYjUwMjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._YAufM9wzDrGtOdi0PMSX3CHGc-fIURmN3ncOrzvDcg)

Don't see why not. Seems to be working as I would expect.

> 4. Maybe better to change label to Biographical info or Bio? ( like in user )

Changed.

myroslavve commented:
> > website - guess, DC link should be to current website, not author link
> > Should be to the website from the field, but maybe it's better to add it as an option.
> > Same for the email.
> 
> Also about the email - we need to at least add the antispambot function for frontend (it's not a great one, but at least something) https://developer.wordpress.org/reference/functions/antispambot/

So nowadays most of the web crawlers seem to work around the wordpress antispambot method, so implemented antispam using javascript to write email on page load, assuming that majority of web crawlers don't execute js, but it still is possible for them to get the email, so added a warning on editor to tell users alternatives to sharing their email on the site:
![image](https://github.com/user-attachments/assets/e46163e1-2a15-4714-a8e8-da3f73add0d9)

myroslavve committed:
Fix normal links with author email selected being decoded

elzadj commented:
@myroslavve weird how the performance-current fails here, could you please isolate it, and check what's going on? Thanks.

elzadj commented:
@myroslavve ignore the performance tests for now, I'm going to change that one page that randomly breaks, and change the tests iterations number.

Olekrut reviewed (changes requested):
https://jam.dev/c/9737cbf1-344d-4694-b570-c93eb14efa41
Create on master author DC blocks ( some of them with links )
FSE and simple editor have errors and broken blocks.
Just blog pattern with broken blocks as well.
![image](https://github.com/user-attachments/assets/ff9d5385-e444-4c72-9c72-5d6a7777971b)
![image](https://github.com/user-attachments/assets/7b3cde31-4cce-48c6-bfe5-a64749157b58)

Everything is ok if create blocks in this branch

myroslavve committed:
Fix validation errors for old blocks with author DC link

myroslavve commented:
> https://jam.dev/c/9737cbf1-344d-4694-b570-c93eb14efa41 Create on master author DC blocks ( some of them with links ) FSE and simple editor have errors and broken blocks. Just blog pattern with broken blocks as well. ![image](https://private-user-images.githubusercontent.com/88893350/366457065-ff9d5385-e444-4c72-9c72-5d6a7777971b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxNzIzMTIsIm5iZiI6MTcyNjE3MjAxMiwicGF0aCI6Ii84ODg5MzM1MC8zNjY0NTcwNjUtZmY5ZDUzODUtZTQ0NC00YzcyLTljNzItNWQ2YTc3Nzc5NzFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDIwMTMzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1YWI5ZDBiMmVkMjRmNWFhNDUxYThiYzBjOWM2ODU2NzU2ZWUzMjQ3ZDYzOGI3ODdmZDY1YWRhOTk2NTllMmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Nl_uN_17Qq8uo3fjI-yXHdbg2zwuAn-fvX2G6MEezys) ![image](https://private-user-images.githubusercontent.com/88893350/366457253-7b3cde31-4cce-48c6-bfe5-a64749157b58.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxNzIzMTIsIm5iZiI6MTcyNjE3MjAxMiwicGF0aCI6Ii84ODg5MzM1MC8zNjY0NTcyNTMtN2IzY2RlMzEtNGNjZS00OGM2LWJmZTUtYTY0NzQ5MTU3YjU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDIwMTMzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxNDNhMDM1YWQ5YmE4ZDA5ZGI5NDIwMGY2MTA3NzU0MmEzMmNlNmY3MGRkYTc2NTE5ZjhhZDljM2E5NTQyYjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9VSvQqOiVmZzIYzok3E_I_zN_Po2ed8ZJEDMQkinJn0)
> 
> Everything is ok if create blocks in this branch

Fixed validation errors on editor, but I couldn't reproduce the errors from php on frontend, if you still have them after this update, could you share code editor with me please.

elzadj commented:
@Olekrut

Olekrut reviewed (changes requested):
1) FSE - 
[FSE DC author.txt](https://github.com/user-attachments/files/17044177/FSE.DC.author.txt)

![image](https://github.com/user-attachments/assets/dc4e50b3-96cf-4b76-9f7c-0624f9ec9eae)

2) only this branch
![image](https://github.com/user-attachments/assets/da7aca8c-0e14-4fae-9694-5083f2760ddd)

myroslavve committed:
Merge branch 'master' into issue-5351

myroslavve committed:
Fix deprecated id attribute warning

myroslavve committed:
Fix author links not working on FSE frontend

myroslavve commented:
> 1. FSE -
>    [FSE DC author.txt](https://github.com/user-attachments/files/17044177/FSE.DC.author.txt)
> 
> ![image](https://private-user-images.githubusercontent.com/88893350/368573396-dc4e50b3-96cf-4b76-9f7c-0624f9ec9eae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgwNjg3MDYsIm5iZiI6MTcyODA2ODQwNiwicGF0aCI6Ii84ODg5MzM1MC8zNjg1NzMzOTYtZGM0ZTUwYjMtOTZjZi00Yjc2LTlmN2MtMDYyNGY5ZWM5ZWFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA0VDE5MDAwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5M2Q4OGY3Y2I0NmRjZTEzY2U0Y2RjNDA2YzUyMTgxODgyNGQ3MjBhYTJkOTNmZjlhMjI3NTgxYjZmYzQyMmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.X6lfkQRh0ufx4wEy_3JD8KHDdq7Fc27GwHGJ3WIC_P8)

Fixed.

> 2. only this branch
>    ![image](https://private-user-images.githubusercontent.com/88893350/368584053-da7aca8c-0e14-4fae-9694-5083f2760ddd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgwNjg3MDYsIm5iZiI6MTcyODA2ODQwNiwicGF0aCI6Ii84ODg5MzM1MC8zNjg1ODQwNTMtZGE3YWNhOGMtMGUxNC00ZmFlLTk2OTQtNTA4M2YyNzYwZGRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA0VDE5MDAwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkZWExY2MyOWI1MTY2YjJlNzdkMmNlMzI5ZTE2MTZlMTJjMTUzMzdiYWMzNWMzZWUyNzVmNjEyZDY3ZTM1NTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ewO8RG-qcRALpr-nIStCcs-NkhVLpQpxEqfOSag02mg)

This error and warnings are all from Jam, was testing and didn't get any of them once. Maybe try to update Jam. I am 95% sure that changes on this branch shouldn't break anything for Jam.

elzadj commented on core/class-maxi-dynamic-content.php:
```diff
@@ -904,20 +904,14 @@
-                ) {
-                    $link = get_author_posts_url($attributes['dc-author']);
+            $dc_link_target = $attributes['dc-link-target'];
+            if (!empty($post) && isset($post->ID)) {
+                if ($dc_link_target === 'author_email') {
+                    $link = $this->xor_obfuscate_email(get_the_author_meta('user_email', $post->ID));
+                } elseif ($dc_link_target === 'author_site') {
+                    $link = get_the_author_meta('user_url', $post->ID);
                 } else {
-                    $link = get_author_posts_url($attributes['dc-id']);
+                    $link = get_author_posts_url($post->ID);
```
that's indeed a little confusing, maybe re-assign the $post variable here, and create another one for $post->ID)

elzadj reviewed (changes requested):
Almost there, good job!

1. The character limit doesn't work on frontend (checked for the
Biographical Info and Full name, but I guess need to check for everything else too)
2. Current's item data option is acceptable on usual pages, need to hide it there.
3. Open in new tab doesn't make sense for the email links.

myroslavve committed:
Disable `no-plusplus` eslint rule

myroslavve committed:
Improve code readability

Put $post->ID in a variable to clarify which id it is, and check if user is WP_User instead of whether it has user data set

myroslavve committed:
Remove link target dropdown for customers

myroslavve committed:
Fix DC limit on frontend for author type

myroslavve committed:
Implement user fields in post/pages on editor

myroslavve committed:
Implement user fields on post/page on frontend

myroslavve committed:
Disable "open in new tab" for author email

myroslavve commented:
> Almost there, good job!
> 
> 1. The character limit doesn't work on frontend (checked for the
>    Biographical Info and Full name, but I guess need to check for everything else too)

Should be fixed.

> 2. Current's item data option is acceptable on usual pages, need to hide it there.

Added author fields to posts/pages types so we can select current author there, like we discussed in chat. Removed current option from "Author" type (only left when editing author archives). @Olekrut could you check this changes, if this is working properly and UI makes sense.
![image](https://github.com/user-attachments/assets/cb0be47f-ac5b-4e79-ab55-31333f3579c7)

> 3. Open in new tab doesn't make sense for the email links.

Disabled it for author email.

myroslavve committed:
Merge branch 'master' into issue-5351

myroslavve committed:
Small improvements

Improved type safety and logic

Olekrut reviewed (changes requested):
Cool) a few things:

1) guess, need to add a field - static text to author list as well
![image](https://github.com/user-attachments/assets/b0aba226-7440-41fb-a083-39a6ddaa7fa8)
2) type customer > image block > what is should be? ( user's avatar or need to hide? )
![image](https://github.com/user-attachments/assets/2fa78212-bc9c-4462-a8b2-6d0963bfd413)
3) type customer > Country ( billing and shipping )> should we  have ability to see not only short name of country ( guess, yes )
![image](https://github.com/user-attachments/assets/34af7552-b3c1-4f74-ae82-a9513a8d7dca)
4) need to add DC email link for billing email 
![image](https://github.com/user-attachments/assets/e41dd8d3-01c5-438d-a325-89cc1c5f38e6)
5) about FSE - get by current archive's item > need to add it to CL for Author archive at least ( guess, because it values are not changing at that moment ) 
![image](https://github.com/user-attachments/assets/2283b131-fe80-4fac-bd78-ef95676daaaa)
6) FSE > type Customer > need to add - get by current archive's item
![image](https://github.com/user-attachments/assets/a7676920-1248-4429-a920-0976298342b1)

myroslavve committed:
Add static text option to author type in DC

myroslavve committed:
Remove customers from types selectable in image DC

myroslavve committed:
Add full county name field to customer type DC

myroslavve committed:
Add customer email link option to DC link

myroslavve committed:
Fix content being single space for undefined name

myroslavve committed:
Enable "current" relation for CL

myroslavve committed:
Merge branch 'master' into issue-5351

myroslavve commented:
> Cool) a few things:
> 
> 1. guess, need to add a field - static text to author list as well

added

> 2. type customer > image block > what is should be? ( user's avatar or need to hide? )

Hid it for now, will add it if we need it in the future.

> 3. type customer > Country ( billing and shipping )> should we  have ability to see not only short name of country ( guess, yes )

Added fields called full country.

> 4. need to add DC email link for billing email

added

> 5. about FSE - get by current archive's item > need to add it to CL for Author archive at least ( guess, because it values are not changing at that moment )

enabled current for CL, checked, it seems to work right, but it was disabled because of https://github.com/maxi-blocks/maxi-blocks/pull/5528#pullrequestreview-1966580790 first point, so could you check it again pls @Olekrut.

> 6. FSE > type Customer > need to add - get by current archive's item
![image](https://github.com/user-attachments/assets/a7676920-1248-4429-a920-0976298342b1)

Customers are not authors so shouldn't be there, but I will create an issue to add current option to customers.

Olekrut reviewed (changes requested):
1) cant to add link to static text
https://jam.dev/c/cdc032d5-181a-4f45-8e21-bf448e41c3ae
2) need to hide it for customer email as well
![image](https://github.com/user-attachments/assets/bd570cce-8372-4351-93c0-7509a08bf817)


Everything else is ok for me, Thank you, Great job!

myroslavve committed:
Allow DC links for static text field

myroslavve committed:
Disable open in new tab for customer email link

myroslavve committed:
Merge branch 'master' into issue-5351

Olekrut reviewed (changes requested):
https://jam.dev/c/f7c99435-e666-42a3-9e2a-020604329ede

cant to add link to static text yet

after last fix > link is disappearing for static text

myroslavve committed:
Reverse allow DC links for static text

custom formats don't work for DC links currently

myroslavve commented:
> https://jam.dev/c/f7c99435-e666-42a3-9e2a-020604329ede
> 
> cant to add link to static text yet
> 
> after last fix > link is disappearing for static text

So DC link doesn't work with custom formats currently, so it was disabled for static text, created a separate issue to allow DC link for static text: https://github.com/maxi-blocks/maxi-blocks/issues/5809

elzadj reviewed (changes requested):
Looking good, just a final push!

- let's keep the nopluplus rule
- In CL, when Bio info is missing, it gets stuck on the previous content instead of 'No content found' 
![Edit-Page-‹-MaxiBlocks-—-WordPress-11-20-2024_02_49_PM](https://github.com/user-attachments/assets/22c99f31-7782-4487-9a41-848324596dd6)
- please add additional checks that rabbit ai asks for
- also pull master, there are some important changes, but they should not affect this PR (hopefully)
- I see a list of site's registered users, not Woo Customers here:
![Edit-Page-‹-MaxiBlocks-—-WordPress-11-20-2024_03_23_PM](https://github.com/user-attachments/assets/53a5d987-316c-4477-8b57-db95ca0dd3c1)

myroslavve committed:
Enable no-plusplus rule

myroslavve committed:
Merge branch 'master' into issue-5351

myroslavve committed:
Only allow customers to be selected for 'customer' type

myroslavve committed:
Sanitize emails + remove code duplication

myroslavve committed:
Get the default field from current field options instead of constant field options

myroslavve commented on src/components/dynamic-content/index.js:
```diff
@@ -637,15 +648,37 @@ const DynamicContent = props => {
 											changeProps({
 												'dc-field':
-													fieldOptions[contentType][
+													fieldOptions[contentType][ // TODO: Check if this will work with currentFieldOptions
```
checked now 😅

myroslavve commented:
> * In CL, when Bio info is missing, it gets stuck on the previous content instead of 'No content found'
>   ![Edit-Page-‹-MaxiBlocks-—-WordPress-11-20-2024_02_49_PM](https://private-user-images.githubusercontent.com/2401618/388100576-22c99f31-7782-4487-9a41-848324596dd6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI5MjIwNjIsIm5iZiI6MTczMjkyMTc2MiwicGF0aCI6Ii8yNDAxNjE4LzM4ODEwMDU3Ni0yMmM5OWYzMS03NzgyLTQ0ODctOWE0MS04NDgzMjQ1OTZkZDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjlUMjMwOTIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDJmZjgwZGEwMTZiMzAyNzMxOTE3YTU5YzllZWVjMGVmMmZkOWJmOTRiOTExYTg2ODM1YmYwZjkwMWY4MjljNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.uvkJxPTMKN3e9I7yrHuHuV91cBjbraGS5rTXwoin2Vs)

I cannot reproduce this, always returns to "No content found" for me, could you share more details please?

Everything else should be fixed. 👍

myroslavve committed:
Improve code clarity

myroslavve committed:
Fix post author fields defaulting to name

Olekrut commented:
this type is empty - for old page
[dc improvements1.txt](https://github.com/user-attachments/files/17993541/dc.improvements1.txt)
![image](https://github.com/user-attachments/assets/c3462fbb-882c-477c-be84-d08692789a87)
![image](https://github.com/user-attachments/assets/238860db-fdd2-420c-beae-82b61e6f48c8)

Olekrut reviewed (changes requested):
this type is empty - for old page
[dc improvements1.txt](https://github.com/user-attachments/files/17993541/dc.improvements1.txt)
![image](https://github.com/user-attachments/assets/2888a1d1-e403-4741-bc4d-d44462ccb2bc)


Error
![image](https://github.com/user-attachments/assets/c3462fbb-882c-477c-be84-d08692789a87)
![image](https://github.com/user-attachments/assets/238860db-fdd2-420c-beae-82b61e6f48c8)

myroslavve commented:
> this type is empty - for old page [dc improvements1.txt](https://github.com/user-attachments/files/17993541/dc.improvements1.txt) ![image](https://private-user-images.githubusercontent.com/88893350/391983448-2888a1d1-e403-4741-bc4d-d44462ccb2bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyNDUxMjgsIm5iZiI6MTczMzI0NDgyOCwicGF0aCI6Ii84ODg5MzM1MC8zOTE5ODM0NDgtMjg4OGExZDEtZTQwMy00NzQxLWJjNGQtZDQ0NDYyY2NiMmJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAzVDE2NTM0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0ZDQwNTMzYTFkNjljNGY5YmY1ZmI1NjU0NWZmNGVlMTk1MmFjZTk4YjFjMjUzYjg2M2NhZmQyYjZlYzk2Y2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7kVsJ0bKHKDPV5lsiQAddxssbE_COYBIfb0IfT66QqE)
> 
> Error ![image](https://private-user-images.githubusercontent.com/88893350/391982892-c3462fbb-882c-477c-be84-d08692789a87.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyNDUxMjgsIm5iZiI6MTczMzI0NDgyOCwicGF0aCI6Ii84ODg5MzM1MC8zOTE5ODI4OTItYzM0NjJmYmItODgyYy00NzdjLWJlODQtZDA4NjkyNzg5YTg3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAzVDE2NTM0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzZmVjZmQ5Nzg3NTNjOGE4OGY1YzNhZTU3NDhkOTM5ZjBmNzkxMGJjYzIwNWYxOTg4YjA2Njc3YTI3OTUxY2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.f-Vpf0SnDJkC0udcbQO-86ZPfMY_1DiDXvLtg9uBK7g) ![image](https://private-user-images.githubusercontent.com/88893350/391982912-238860db-fdd2-420c-beae-82b61e6f48c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyNDUxMjgsIm5iZiI6MTczMzI0NDgyOCwicGF0aCI6Ii84ODg5MzM1MC8zOTE5ODI5MTItMjM4ODYwZGItZmRkMi00MjBjLWJlYWUtODJiNjFlNmY0OGM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAzVDE2NTM0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwYmI5MTNlNWM0NWQ3NmRiMDZiZjVmMDkzZWJmODhiZjViMTk5Nzk4NTE2ZDg3MDc5MjRkN2FjYmZmN2JiMmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fUUi-qeEU-XHIxGxE-qx0UlBQX0b7lrvymEXLFQICvQ)

I can't reproduce the error or the "This type is empty" problem, tried on old pages on my local, and created a page on master with the code editor you have provided, both work good for me. Maybe there is something I am missing, if you could record a video in jam would be good.

Olekrut reviewed (changes requested):
@elzadj  can re produce it as well with file from previous comment, so
https://jam.dev/c/7e500804-0a01-4088-b936-a9b4bfd5cb0d

myroslavve committed:
Fix "This type is empty" problem for posts with customer type DC

Olekrut reviewed (changes requested):
Change author in CL > click to some block > open CL again > author is backing to first ( automatically )
https://jam.dev/c/8962452b-3864-4343-ae30-c408d7963889

The same with new blocks
https://jam.dev/c/f2a5a7ff-4563-4c6d-b397-772fe407ffda

myroslavve committed:
Fix author resetting every time CL settings are opened

elzadj reviewed (changes requested):
The dropdown with woo customers is not working for me:

[Add-New-Page-‹-MaxiBlocks-—-WordPress (1).webm](https://github.com/user-attachments/assets/0a5db868-61ba-4f3b-ac70-c28e4068c379)

myroslavve committed:
Fix current author being used as fallback for customers

When there are no customers on a site `getDCContent` returned author of the post as a customer

myroslavve committed:
Update id list to an empty one when no entities found

myroslavve commented:
> The dropdown with woo customers is not working for me:
> 
>  Add-New-Page-.-MaxiBlocks-.-WordPress.1.webm

Seems like the problem is happening when there are no customers on the site, at least that's how I could reproduce it.
Fixed for this case, if there are any other cases when it happens, let me know.

elzadj reviewed (changes requested):
Getting empty. I have 2 customers with completed orders on the site.

![Add-New-Page-‹-MaxiBlocks-—-WordPress-12-10-2024_05_33_PM](https://github.com/user-attachments/assets/7278c91d-ba95-46f1-9437-0ab2dbc238b5)

myroslavve committed:
Add "current" relation for customers and remove all other relations

myroslavve commented:
For now, made "current" the only available relation option for customers.

elzadj reviewed (changes requested):
Backend and frontend both look weird (my current user is a customer):

[Edit-Page-‹-MaxiBlocks-—-WordPress (3).webm](https://github.com/user-attachments/assets/2ddbf3df-4b8b-4548-8cce-1a14ecffceae)

![MaxiBlocks-12-13-2024_11_57_AM](https://github.com/user-attachments/assets/bd6bf3af-4071-430b-bd01-09cbe1d0093b)

Maybe let's just remove customers?

myroslavve committed:
Remove "customers" DC type

myroslavve committed:
Remove redundant import

myroslavve commented:
> Backend and frontend both look weird (my current user is a customer):
> 
>  Edit-Page-.-MaxiBlocks-.-WordPress.3.webm 
> ![MaxiBlocks-12-13-2024_11_57_AM](https://private-user-images.githubusercontent.com/2401618/395515232-bd6bf3af-4071-430b-bd01-09cbe1d0093b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNTY1NTAsIm5iZiI6MTczNDM1NjI1MCwicGF0aCI6Ii8yNDAxNjE4LzM5NTUxNTIzMi1iZDZiZjNhZi00MDcxLTQzMGItYmQwMS0wOWNiZTFkMDA5M2IucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTMzNzMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTVlMzhlZDcxZDU1YTg2Y2FjMWU3ZjJjZDY1NzRmMjBkZjU1MDM5ZTgwY2I3M2RkMjAxYTFhNDRmMTZlNTE4ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.daFDisWJwR_qWGDVsHe7hJWfe30tRlhph7YF0C1L_Ko)
> 
> Maybe let's just remove customers?

Yes, removed them for now, it would be hard to fix your issue, so better do it if we will be implementing customers fully in another issue.

elzadj commented:
Looks good
@Olekrut check pls, and then we merge
